lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -582,6 +583,38 @@ void TypeLoweringVisitor::visitDecl(RegOp op) {
   opsToRemove.push_back(op);
 }
 
+/// Lower a RegReset op with a bundle to multiple non-bundled RegResets.
+void TypeLoweringVisitor::visitDecl(RegResetOp op) {
+  Value result = op.result();
+
+  // Attempt to get the bundle types, potentially unwrapping an outer flip type
+  // that wraps the whole bundle.
+  FIRRTLType resultType = getCanonicalAggregateType(result.getType());
+
+  // If the RegReset is not a bundle, there is nothing to do.
+  if (!resultType)
+    return;
+
+  SmallVector<FlatBundleFieldEntry, 8> fieldTypes;
+  flattenType(resultType, "", false, fieldTypes);
+
+  // Loop over the leaf aggregates.
+  for (auto field : fieldTypes) {
+    SmallString<16> loweredName(op.nameAttr().getValue());
```

Comment:
I believe the name is actually an optional attribute, so you have to check if it has a value before using it.  I think the RegOp has this same issue.

Reply:
@youngar , Thanks for catching this, confirmed it crashes when the name is not present. Will create a new issue with this